Dockerfile

```diff
@@ -1,6 +1,11 @@
-FROM ruby:2.3.7
+FROM ruby:2.3.7-slim
 ENV PHANTOM_JS=2.1.1
 
+# Add basic binaries and clean up the apt cache
+RUN apt-get update \
+  && apt-get install -y bzip2 curl gnupg wget \
```

Comment:
`apt-get install -y bzip2 curl gnupg wget`

These are super common, fundamental utilities that lots of stuff relies on. Probably need these by the end of the script. However, it'slikely these are provided during the `yarn` or `nodejs` install scripts. That makes this line redundant, and `apt-get` chews up time during `docker-compose`. So IMO we should test removing this if we want the speediest docker-compose times.

- `bzip2`: compression/decompression algorithm
- `curl`, `wget`: tiny utilities to download stuff off of the network
- `gnupg`: GNU Privacy Guard implements pgp cryptographic signing, helps verify stuff you downloaded hasn't been tampered with

Reply:
Yeah, I think it would be best to remove everything but the change in image and see if it works from there.

Reply:
I realized just now we have installer scripts for `yarn` and `nodejs` thatwe download with `curl`. Likewise, we currently download the PhantomJS prebuilt package with `wget`. So:
- `curl` and `wget` need to stay.
  - OR we should have them both use `curl`, or both use `wget` and eliminate the download utility we don't use.

(Aside: to confirm this, I tried without `curl` and the build failed. I can try again without `wget`, but it's a bit tedious. The concept of this slim ruby base image is that it *doesn't* include most of these common utilities, whereas the regular ruby base image does include the super-common ones.)

I assume we do use `bzip2` and `gnupg`, but I'll try without. be back in a bit with the results.

Reply:
So it looks like we need all four of these packages in order to install with the slim base image.

The rm `/var/lib/apt/lists/*` is totally optional, and just decreases the size of the built Docker image.